internal/errors/errors_test.go

```diff
@@ -0,0 +1,1498 @@
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			wantErr := errors.New("server gateway.vald.svc.cluster.local not found")
+			return test{
+				name: "return an ErrServerNotFound error when the name is not empty.",
+				args: args{
+					name: "gateway.vald.svc.cluster.local",
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("server  not found")
+			return test{
+				name: "return an ErrServerNotFound error when the name is empty.",
+				args: args{
+					name: "",
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrServerNotFound(test.args.name)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrOptionFailed(t *testing.T) {
+	type args struct {
+		err error
+		ref reflect.Value
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			fmt.Println(got)
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			wantErr := errors.New("failed to setup option :\tfmt.Println: option failed error")
+			return test{
+				name: "return an ErrOptionFailed error when err and ref are not empty.",
+				args: args{
+					err: errors.New("option failed error"),
+					ref: func() reflect.Value {
+						var i interface{} = fmt.Println
+						return reflect.ValueOf(i)
+					}(),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("failed to setup option :\tfmt.Println")
+			return test{
+				name: "return an ErrOptionFailed error when err is empty and ref is not empty.",
+				args: args{
+					ref: func() reflect.Value {
+						var i interface{} = fmt.Println
+						return reflect.ValueOf(i)
+					}(),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("failed to setup option :\t: option failed error")
+			return test{
+				name: "return an ErrOptionFailed error when err is not empty and ref is empty.",
+				args: args{
+					err: errors.New("option failed error"),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
```

Comment:
can you add a test case that both `ref` and `err` are empty?

Reply:
I already added